### PR TITLE
Add docs for init and install command, describe declarative examples, change repo-revision, change trigger for documentation deployment

### PR DESCRIPTION
**Pull request by Rohitrajak1807**
Fix: #360 
Fix: #325 
Fix: #375 

@cruizen  This PR also changes the trigger for mkdocs github action and when rebasing from `main`, I discovered that 

`k8s.io/client-go v0.23.14` changes to `k8s.io/client-go v11.0.1-0.20190816222228-6d55c1b1f1ca+incompatible` in go.mod

### CODE REVIEW

**Review by ShaunakJoshi1407 — APPROVED** [review]

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+          token: ${{ secrets.GH_PAT }}
+          check-latest: true
+          cache: 'pip' # caching pip dependencies
+          cache-dependency-path: '**/requirements.txt'
+      - run: pip install -r docs/requirements.txt
+      - run: mkdocs gh-deploy --force
```
Comment (cruizen) [inline .github/workflows/docs.yaml:30]: I think we should just skip the deployment for docs for now. We will have to replace `mkdocs deploy` with `mike deploy` but looks like that requires additional work.

File: docs/installation.md [hunk 2]
```diff
@@ -84,10 +95,15 @@ The following instructions are to manually build CLI from this code repository.
 ## Cluster orchestration API providers
 
 Arlon currently supports Cluster API on AWS cloud. It also has experimental
-support for Crossplane on AWS.
+support for Crossplane on AWS. It `cluster-api` cloud provider components on
```
Comment (cruizen) [inline docs/installation.md:98]: > It `cluster-api` cloud provider components. 
Did not follow this.

File: docs/installation.md [hunk 3]
```diff
@@ -119,15 +144,71 @@ I0222 17:31:14.112689   27922 request.go:668] Waited for 1.046146023s due to cli
 
 Arlon CLI provides an `init` command to install "itself" on a management cluster.
 This command performs a basic setup of `argocd`(if needed) and `arlon` controller.
-If `argocd` is already installed, it assumes that `admin` password is the same as in `argocd-initial-admin-secret` ConfigMap and that `argocd` resides in the `argocd` namespace.
-Similar assumptions are made for detecting Arlon installation as well: assuming that the existence of `arlon` namespace means Arlon controller exists.
-To install Arlon controller using the init command these pre-requisites need to be met:
+It makes the following assumptions while installing `arlon`:
+
```
Comment (cruizen) [inline docs/installation.md:148]: It makes the following assumptions while installing `arlon`:

something missing here?

File: docs/installation.md [hunk 4]
```diff
@@ -119,15 +144,71 @@ I0222 17:31:14.112689   27922 request.go:668] Waited for 1.046146023s due to cli
...
+1. A valid kubeconfig pointing to the management cluster.
+1. Port **8080** should not be in use by other programs. Arlon init uses it to port-forward `argocd`.
+1. A hosted Git repository with at least a `README` file present and a valid GitHub token([detailed here](#setting-up-the-workspace-repository)) for:
+  1. Adding a repository to ArgoCD.
+  1. Avoiding rate limiting of GitHub API while fetching `cluster-api` related manifests.
+  1. Pushing cluster template manifests to the workspace repository.
+1. Pre-requisites for supported CAPI infrastructure providers(AWS and Docker as of now) as [described below](#pre-requisites-for-cluster-api-providers).
+
+## Setting up the workspace repository
```
Comment (cruizen) [inline docs/installation.md:166]: `markdownlint` (integrated in VS code) requires an empty line after a heading.

**Review by cruizen — CHANGES_REQUESTED** [review]

File: .github/workflows/docs.yaml (hunk 1 above)
Comment (Rohitrajak1807) [reply .github/workflows/docs.yaml:30]: done. this file is deleted

File: docs/installation.md (hunk 3 above)
Comment (Rohitrajak1807) [reply docs/installation.md:149]: done.

File: docs/installation.md (hunk 2 above)
Comment (Rohitrajak1807) [reply docs/installation.md:99]: done

**Review by cruizen — APPROVED** [review]

**Pull request merged**